samples/sid_end_device/include/sidewalk.h

```diff
@@ -9,30 +9,79 @@
 #include <zephyr/sys/slist.h>
 #include <sid_api.h>
 #include <zephyr/kernel.h>
+#include <zephyr/smf.h>
 
-typedef enum {
-	SID_EVENT_SIDEWALK,
-	SID_EVENT_FACTORY_RESET,
-	SID_EVENT_NEW_STATUS,
-	SID_EVENT_SEND_MSG,
-	SID_EVENT_CONNECT,
-	SID_EVENT_LINK_SWITCH,
-	SID_EVENT_NORDIC_DFU,
-	SID_EVENT_FILE_TRANSFER,
-	SID_EVENT_LAST,
-} sidewalk_event_t;
-
-typedef struct {
-	sidewalk_event_t id;
-	void *ctx;
-} sidewalk_ctx_event_t;
+enum sidewalk_fsm_states {
+	STATE_SIDEWALK,
+	STATE_DFU,
 
+	STATE_LAST
+};
 typedef struct {
 	struct sid_handle *handle;
 	struct sid_config config;
 	struct sid_status last_status;
 } sidewalk_ctx_t;
 
+typedef struct {
+	const char *name;
+	void (*call[STATE_LAST])(void *ctx, void *state_machine);
+} sidewalk_event_t;
+typedef struct {
+	sidewalk_event_t event_handler;
+	void *ctx;
+} sidewalk_ctx_event_t;
+
+typedef struct sm_s {
+	struct smf_ctx ctx;
+	struct smf_state sid_states[STATE_LAST];
```

Comment:
Why is a whole table of states needed byt he state machine struct now?

Reply:
is is easier to access the states for transition. 
previously the event handlers were in the same file as the global state table, now they can be anywhere.
Because I pass the `sm_t` struct to the event handler, it is easy to access states for transition no matter where the actual hanlder is placed.